module/os/linux/zfs/zfs_znode.c

```diff
@@ -2127,8 +2127,10 @@ zfs_obj_to_path_impl(objset_t *osp, uint64_t obj, sa_handle_t *hdl,
 		size_t complen;
 		int is_xattrdir = 0;
 
-		if (prevdb)
+		if (prevdb) {
+			ASSERT(prevhdl != NULL);
```

Comment:
Adding the ASSERT makes good sense.  Can you just also add it to the matching function in `module/os/freebsd/zfs/zfs_znode.c` and force update the PR.

Reply:
Hi @behlendorf! Thanks. Yeah, I did it.